lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -880,6 +881,19 @@ ParseResult FIRParser::parseType(FIRRTLType &result, const Twine &message) {
     if (parseEnumType(result))
       return failure();
     break;
+
+  case FIRToken::kw_const: {
+    consumeToken(FIRToken::kw_const);
+    if (failed(parseType(result, message)))
+      return failure();
+
+    auto baseType = result.dyn_cast<FIRRTLBaseType>();
+    if (!baseType)
+      return emitError("only hardware types can be 'const'");
+
+    result = baseType.getConstType(true);
+    return success();
```

Comment:
Does this let you parse `const const const const {}`?

Reply:
Yeah, should probably fix it.

Reply:
Updated